src/ProjectTemplates/test/ProjectExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Templates.Test.Helpers;
+
+namespace Templates.Test;
+
+public static class ProjectExtensions
+{
+    public static async Task<Project> VerifyLaunchSettings(this Project project, string[] expectedLaunchProfileNames)
+    {
+        var launchSettingsFiles = Directory.EnumerateFiles(project.TemplateOutputDir, "launchSettings.json", SearchOption.AllDirectories);
+
+        foreach (var filePath in launchSettingsFiles)
+        {
+            using var launchSettingsFile = File.OpenRead(filePath);
+            using var launchSettings = await JsonDocument.ParseAsync(launchSettingsFile);
+
+            var profiles = launchSettings.RootElement.GetProperty("profiles");
+            foreach (var profileName in expectedLaunchProfileNames)
+            {
+                Assert.True(profiles.TryGetProperty(profileName, out var _), $"Expected launch profile '{profileName}' was not found in file {filePath}");
+            }
+
+            if (launchSettings.RootElement.TryGetProperty("iisSettings", out var iisSettings)
+                && iisSettings.TryGetProperty("iisExpress", out var iisExpressSettings))
+            {
+                var iisSslPort = iisExpressSettings.GetProperty("sslPort").GetInt32();
+                if (expectedLaunchProfileNames.Contains("https"))
+                {
+                    Assert.True(iisSslPort > 44300, $"IIS Express port was expected to be greater than 44300 but was {iisSslPort} in file {filePath}");
```

Comment:
Is this supposed to be >= 44300?  I see the mac test failure with `IIS Express port was expected to be greater than 44300 but was 44300 in file`

Reply:
Altho that makes me wonder whether we need to run iisExpress tests on mac?

Reply:
Yeah I was going to change that to >= 44300 after seeing the failure although I want to re-check the template logic to see if it's valid that this number can be 44300 after the template's port-replacement logic has run. The template doesn't appear to have any logic to suppress adding the IIS Express settings on non-Windows machines so it's always there but we could change this method to only do that part of the verification if the platform is Windows.

Reply:
Well I guess if the templates always do it regardless of platform, we should just test it that way even if it feels weird